### PR TITLE
Role Module

### PR DESCRIPTION
**Pull request by samargupta007**

### CODE REVIEW

File: lib/business/roleModule.mjs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                throw new Error('RoleId and ModuleId are required.');
+            }
+        }
+
+        // Remove Permission1, Permission2, etc., and combine them into the Permissions field
+
+
+        // Create a permissions binary string from the individual Permission fields
+        const permissionsBinary = permissionFields
+            .map(field => req.body[field] === '1' || req.body[field] === true ? '1' : '0') // Check if field is '1' or '0'
```
Comment (Pramodspraxa) [inline lib/business/roleModule.mjs:26]: you should add condition in parenthesis - 
(req.body[field] === '1' || req.body[field] === true) ? '1': '0'

File: lib/business/roleModule.mjs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+
+    // Method to handle any processing after loading RoleModule data
+    static afterLoad = async ({ data = {} }) => {
+        // If Permissions exist, transform related fields
+
+        // Convert Permission fields from 1/0 to true/false
+        permissionFields.forEach(field => {
+            if (field in data) {
+                data[field] = data[field] === 1 ? true : false; // Convert to boolean
```
Comment (Pramodspraxa) [inline lib/business/roleModule.mjs:46]: you can avoid using true /false

File: lib/business/roleModule.mjs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        const permissionsBinary = permissionFields
+            .map(field => req.body[field] === '1' || req.body[field] === true ? '1' : '0') // Check if field is '1' or '0'
+            .join(''); // Join them into a single string
+
+        // Update the Permissions field with the combined binary string
+        req.body.Permissions = permissionsBinary;
+        delete req.body.Name;
+
+        // Remove the individual Permission fields from the request body
+        permissionFields.forEach(field => delete req.body[field]);
```
[the remark this answers is not in the document]
Comment (samargupta007) [reply lib/business/roleModule.mjs:31]: added the return statement

File: lib/business/roleModule.mjs (hunk 2 above)
Comment (samargupta007) [reply lib/business/roleModule.mjs]: updated

**Review by manojpandeyDev18 — APPROVED** [review]

**Pull request merged**